added improved jupyter notebook and corresponding csv files

marvinbuss opened the pull request:
## Purpose
<!-- Added an improved Jupyter Notebook with csv file support and plotting in matplotlib. Also, I added a connection to Power BI to simulate data streaming to Power BI. -->
* ...

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
<!-- Just run the jupyter notebook. -->
```
```

## What to Check
Verify that the following are valid
* Verify the correct functionality of the notebook. I tested it all, but please validate correct functionality.

## Other Information
<!-- Add any other helpful information that may be needed here. -->

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure-Samples/AnomalyDetector?pullRequest=13) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure-Samples/AnomalyDetector?pullRequest=13) before we can accept your contribution.<br/><br/>:x: marvinbuss [sign now](https://cla.opensource.microsoft.com/Azure-Samples/AnomalyDetector?pullRequest=13)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure-Samples/AnomalyDetector?pullRequest=13) it.</sub>

yingqunpku commented:
@TonyXing  and @conhua could you help review Marvin's PR?

yingqunpku commented:
@marvinbuss 
In your notebook, I have noticed send_to_power_bi function. I assume there'll be a proper PowerBI setup needed in order to make that function work (with POWER_BI_REST_API_URL). What is the setup process? Is it possible to have a powerbi desktop project or a readme to guide users have their pbi setup ready to consume the api call?

marvinbuss commented:
@yingqunpku 
You can find a description in my repository (https://github.com/marvinbuss/MS-AnomalyDetector). I can add that to one of the Readme files in your repos.
If you want me to, I can also remove the feature.

yingqunpku commented:
@marvinbuss  a few comments:
1. I'm not able to try out the PBI function as the code fails on my local env 
TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
[error.txt](https://github.com/Azure-Samples/AnomalyDetector/files/3244691/error.txt)

2. In Powerbi, there're 3 options when creating a streaming dataset: api, azure stream, and pubnub. I guess it should go with api the default value but you may want to clarify that with your readme.

3. read_csv("data/sample_hourly.csv") "data" dir does not exist in this repo. You may want to change all those to ../example-data/  OR create a folder and move your data files to the folder.

marvinbuss commented:
@yingqunpku 

1. I was not able to reproduce the error, but other users report the same issue (https://stackoverflow.com/questions/29329725/pandas-and-matplotlib-fill-between-vs-datetime64). I updated the notebook with  the proposed solution (`data['Date'].dt.to_pydatetime()`). This should hopefully fix the issue with some pandas versions.

2. I updated the Readme File in my repository and added more details to the instructions: https://github.com/marvinbuss/MS-AnomalyDetector

3. I sadly comitted the wrong file on Friday. This was already fixed in the first commit.

4. We should delete the JSON files in AnomalyDetector/ipython-notebook/ and update the file names in AnomalyDetector/example-data/. Currently, it is not very well structured and can be improved. We could also keep the files that I uploaded in this branch.
